### PR TITLE
feat: support integer formatting in printf

### PR DESCRIPTION
**Pull request by viveknathani**
## Description

Adds width specifier support for %d and %i format specifiers in the printf function. This continues the support for the issue https://github.com/tursodatabase/turso/issues/885.

Examples:
  - Width specifiers: printf('%5d', 42) → "   42" (right-padded to 5 characters)
  - Zero-padding: printf('%05d', 42) → "00042" (zero-padded instead of spaces)
  - Precision: printf('%.3d', 42) → "042" (minimum digits)
  - Combined: printf('%5.3d', 42) → "  042" (width + precision together)
  
Behavior matches SQLite's printf implementation.

## Motivation and context

I got interested in contributing and I saw in COMPAT.md that `printf` needs better support for modifiers. This felt like a good first contribution to make. It aligns with the goal of making Turso compatible with SQLite.

## Description of AI Usage

Claude code helped me with a code review. It suggested eliminating code duplication by parsing width separately from format type handling.

### CODE REVIEW

File: testing/scalar-functions-printf.test [hunk 1]
```diff
@@ -275,3 +275,17 @@ do_execsql_test printf-mixed-exponential-integer {
 do_execsql_test printf-mixed-octal-hex {
   SELECT printf('Octal %o is hex %x', 255, 255);
 } {{Octal 377 is hex ff}}
+
+# width formatting for integers
+
+do_execsql_test printf-width-integer-basic {
```
Comment (Pavan-Nambi) [inline testing/scalar-functions-printf.test]: Hey!, thanks for the contribution!

could u please add more test cases? like when we pass negative.. or 0 or something stupid..etc
for eg: `01d` , `-1d` , `0.1d` ...etc..

Comment (viveknathani) [reply testing/scalar-functions-printf.test]: hey! added these. cases like `0d` and precision-only formats such as `.1d` are now explicitly handled and covered by tests.

i’ve kept formats like `-1d` out of scope for this PR. negative widths imply left-alignment, and the `-` flag requires its own handling. felt better to keep this change focused and leave negative-width handling for a follow-up.

Comment (Pavan-Nambi) [reply testing/scalar-functions-printf.test]: yeah that works just fine, thanks!

**Comment by viveknathani** [comment]
hi @pereman2 - quick ping, would appreciate your thoughts when you get a chance.

**Comment by viveknathani** [comment]
hi folks, doing a quick bump, happy to make any changes if needed - @pereman2 @jussisaurio

**Comment by LeMikaelF** [comment]
@claude fix the conflicts

@viveknathani sorry about the delay, I'll get to reviewing this after claude has fixed the merge conflict.

**Comment by LeMikaelF** [comment]
@viveknathani There's a few functional issues with this PR. I rebased it and added tests, some of which are failing. Could you address them?

**Comment by viveknathani** [comment]
hi @LeMikaelF - firstly, thank you for getting back and fixing the conflicts!

i took a look at the failing test after your commit and saw that it was a `sscache` timeout issue in `Rust / test-limbo` which seemed like a thing that should be fixed on a re-run.

so i pushed an empty commit to trigger the CI re-run which led to the following:
- we now have 4 failing tests, and 4 cancelled tests.
- the cancelled tests just have the message "The operation was canceled" after running for about 20 mins
- amongst the 4 failing tests, 3 are of the category `SQL Tests / Run SQL tests` and I saw that these are failing on other PRs too? i also saw that you have a PR to fix these tests?
- the final failing test (`Rust / build-native (blacksmith-4vcpu-ubuntu-2404)`) has the message `The action 'Test (nextest with coverage)' has timed out after 20 minutes.`

**Comment by LeMikaelF** [comment]
@viveknathani my bad, I merged something that broke main yesterday. If you just rebase on https://github.com/tursodatabase/turso/pull/4951 , the SQL tests will pass. Don't worry about the timeouts, our CI is going through a rough patch (temporarily).

**Comment by viveknathani** [comment]
gotcha @LeMikaelF - just rebased on main since the referenced change got merged. the tests are passing now.

**Comment by LeMikaelF** [comment]
@viveknathani I'm confused, it looks like maybe you force-pushed this to an earlier stage? The test file is now a `.test` file (our legacy format), but the tests I added were to a `sqltest` file (the new and preferred format).

**Comment by Pavan-Nambi** [comment]
i think you should restore 'db' files to there previous state, i'm unsure why they are getting modified tho, i haven't seen that in a while.

**Comment by viveknathani** [comment]
@LeMikaelF - my bad, had messed it up. fixed it now, using latest changes from `main`. and my tests are now just additions to the existing `.sqltest` file. please take a look now! 🫡

@Pavan-Nambi - thanks! restored the state of those .db files

**Comment by LeMikaelF** [comment]
@viveknathani I just pushed another commit with tests, there's still some over/underflows left that panic in debug builds.

**Comment by LeMikaelF** [comment]
Thanks @viveknathani !

There were still some panics that my Claude Code found, and I rebased on main because there were concurrent changes in this area. This should get merged tomorrow.

**Review by LeMikaelF — APPROVED** [review]

**Comment by LeMikaelF** [comment]
@claude fix the fuzzer failure

**Comment by LeMikaelF** [comment]
@claude fix the fuzzer failure

@viveknathani looks like our fuzzer is still catching a lot of panics in this code. If you want to keep working on this, tell me, otherwise I'm just going to claude-loop it until it passes all tests.

**Comment by LeMikaelF** [comment]
looks like the claude bot isn't able to push to a fork :(

**Comment by viveknathani** [comment]
thanks @LeMikaelF -  i pushed claude's work and the pipeline is green for us now

**Comment by viveknathani** [comment]
closing this as it’s superseded by #5190.

thanks @LeMikaelF and @Pavan-Nambi for all the reviews and discussions over the past couple of months!

**Comment by glommer** [comment]
hi @viveknathani ! I just want to say thank you for your effort here, and don't get discouraged! I was personally not aware that this PR existed (there are so many!) and we could have merged it, but reality is that the printf machinery needed a complete overhaul and I ended up doing it.

We'd love to see you back!

**Comment by viveknathani** [comment]
thank you for your kind words @glommer!

i totally get it, the rewrite makes sense for the long-term direction.

appreciate the encouragement! 😄
